src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -143,6 +156,14 @@ assign(ReactDOMTextComponent.prototype, {
   },
 
   unmountComponent: function() {
+    var openingComment = this._openingComment;
+    if (openingComment) {
+      DOMChildrenOperations.removeDelimitedText(
```

Comment:
So now I remember the problem with doing that in DOMChildrenOperations was that if I set `_openingComment` to null here, I wouldn't be able to access it in DOMChildrenOperations. Not sure of a good way around that. So I just added a `getOpeningCommentNode` method that takes the closing comment as a param and walks back from there (couldn't look at `this._nativeNode` either since it's also null in DOMChildrenOperations).